subsys/net/ip/net_if.c

```diff
@@ -125,6 +125,46 @@ static sys_slist_t timestamp_callbacks;
 #define debug_check_packet(...)
 #endif /* CONFIG_NET_IF_LOG_LEVEL >= LOG_LEVEL_DBG */
 
+struct net_if *z_impl_net_if_get_by_index(int index)
+{
+	if (index <= 0) {
+		return NULL;
+	}
+
+	if (&__net_if_start[index - 1] >= __net_if_end) {
+		NET_DBG("Index %d is too large", index);
+		return NULL;
+	}
+
+	return &__net_if_start[index - 1];
+}
+
+#ifdef CONFIG_USERSPACE
+struct net_if *z_vrfy_net_if_get_by_index(int index)
+{
+	struct net_if *iface;
+	struct z_object *zo;
+	int ret;
+
+	iface = net_if_get_by_index(index);
+	if (!iface) {
+		return NULL;
+	}
+
+	zo = z_object_find(iface);
+
+	ret = z_object_validate(zo, K_OBJ_NET_IF, _OBJ_INIT_TRUE);
```

Comment:
Can't you use `Z_SYSCALL_OBJ` ? Or the error is recoverable ?

Reply:
If there is an error, the function will return invalid value which in this case is ok. The caller needs to handle invalid values anyway.